src/coldcard.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Provides classes for interacting with a Coldcard via TXT/JSON/PSBT files
+ *
+ * @module coldcard
+ */
+import {
+  IndirectKeystoreInteraction,
+} from "./interaction";
+
+export const COLDCARD = 'coldcard';
+
+export function parseData() {}
+
+export function encodeData() {}
+
+/**
+ * Base class for interactions with Coldcard.
+ *
+ * @extends {module:interaction.IndirectKeystoreInteraction}
+ */
+export class ColdcardInteraction extends IndirectKeystoreInteraction {
+
+
+  /**
+   * Coldcard interactions need to know the bitcoin network they are on
+   *
+   * @param {object} options - options argument
+   * @param {string} options.network - bitcoin network
+   */
+  constructor({network}) {
+    super();
+    this.network = network;
+  }
+}
+
+/**
+ * Base class for interactions
+ *
+ * @extends {module:hermit.HermitInteraction}
+ */
+export class ColdcardFileReader extends ColdcardInteraction {}
+
+export class ColdcardExportExtendedPublicKey extends ColdcardFileReader {}
+
+export class ColdcardSignTransaction extends ColdcardFileReader {}
+
+/**
+ * Returns a valid multisig wallet config text file to send over to a Coldcard
+ *
+ * NOTE: while it will technically work with unknown root fingerprints (xfp's)
+ * we highly recommend only setting up the multisig wallet with complete xfp
+ * information. We define isSupported() to check exactly this condition.
+ *
+ * This is an example Coldcard config file from
+ * https://coldcardwallet.com/docs/multisig
+ *
+ * # Coldcard Multisig setup file (exported from 4369050F)
+ * #
+ * Name: MeMyself
+ * Policy: 2 of 4
+ * Derivation: m/45'
+ * Format: P2WSH
+ *
+ * D0CFA66B: tpubD9429UXFGCTKJ9NdiNK4rC5...DdP9
+ * 8E697B74: tpubD97nVL37v5tWyMf9ofh5rzn...XgSc
+ * BE26B07B: tpubD9ArfXowvGHnuECKdGXVKDM...FxPa
+ * 4369050F: tpubD8NXmKsmWp3a3DXhbihAYbY...9C8n
+ *
+ */
+export class ColdcardConfig {
+  constructor({jsonConfig}) {
+    this.jsonConfig = jsonConfig;
+    //this.jsonConfig.derivation = "m/45'";
+  }
+
+  /**
+   * @returns {boolean} isSupported - does the jsonConfig have necessary pieces?
+   */
+  isSupported() {
```

Comment:
To me this feels like a check (and a rejection) that should happen at the instantiation of a new class object. If when creating a new instance of a `ColdcardConfig` the required information is missing (maybe we're not requiring it and just want to log a warning? not sure how we handle "suggestions" but we're throwing in `translate` if this is missing so we're treating it as required so far) then the class should never be instantiated in the first place. So this to me should be in the constructor imo.